backend/crypt/cipher_test.go

```diff
@@ -194,20 +197,37 @@ func TestDecryptFileName(t *testing.T) {
 		{NameEncryptionStandard, true, "p0e52nreeaj0a5ea7s64m4j72s/l42g6771hnv3an9cgc8cr2n1ng/qgm4avr35m5loi1th53ato71v0", "1/12/123", nil},
 		{NameEncryptionStandard, true, "p0e52nreeaj0a5ea7s64m4j72s/l42g6771hnv3an9cgc8cr2n1/qgm4avr35m5loi1th53ato71v0", "", ErrorNotAMultipleOfBlocksize},
 		{NameEncryptionStandard, false, "1/12/qgm4avr35m5loi1th53ato71v0", "1/12/123", nil},
+		{NameEncryptionStandard, true, "p0e52nreeaj0a5ea7s64m4j72s-v9999-99-99-000000-000", "", base32.CorruptInputError(26)},
```

Comment:
I think I'd rather see some tests here with valid date suffixes too than your extra loop adding the suffixes programatically and checking them I think.

That would mean that if date suffix adding code changes we'll get a warning about it in the tests and we can see explicitly in the tests what we have.